deepchem/models/lightning/dc_lightning_dataset_module.py

```diff
@@ -21,53 +26,114 @@ class DCLightningDatasetModule(L.LightningDataModule):
     """DeepChem Lightning Dataset Module to be used with the DCLightningModule and a Lightning trainer.
 
     This module wraps over the the deepchem pytorch dataset and dataloader providing a generic interface to run training.
-    https://pytorch-lightning.readthedocs.io/en/stable/extensions/datamodules.html
 
     Notes
     -----
-    This class requires PyTorch to be installed.
+    This class requires PyTorch and lightning to be installed.
+    For more information, see:
+      - PyTorch Lightning DataModule Documentation: https://pytorch-lightning.readthedocs.io/en/stable/extensions/datamodules.html
     """
 
     def __init__(self,
                  dataset: dc.data.Dataset,
                  batch_size: int,
                  collate_fn: Callable = collate_dataset_wrapper,
-                 num_workers: int = 0):
+                 num_workers: int = 0,
+                 model: Optional[TorchModel] = None):
         """Create a new DCLightningDatasetModule.
 
         Parameters
         ----------
         dataset: dc.data.Dataset
-            A deepchem dataset.
+            A deepchem loaded dataset.
         batch_size: int
             Batch size for the dataloader.
         collate_fn: Callable
-            Method to collate instances across batch.
+            Custom collate function. Defaults to collate_dataset_wrapper.
         num_workers: int
             Number of workers to load data
+        model: Optional[TorchModel], Defaults to None
+            DeepChem model for collate function, which is used to utilize the model's
+            `_prepare_batch` and `default_generator` method for preparing the batch data.
         """
         super().__init__()
         self._batch_size = batch_size
-        self._dataset = dataset.make_pytorch_dataset(
-            batch_size=self._batch_size)
-        self.collate_fn = collate_fn
+        if model is None:
```

Comment:
Add a comment here explaining this if/else. When is model None, when is it not None?